### PR TITLE
Improving Eloquent Performance

### PR DESCRIPTION
**Pull request by Thotsuya**
I did the best I could by selecting only the fields that were shown in the view, as well as using eager loading and the new withAvg Eloquent method. Not the best but I did what I could by only using Eloquent.

- Time 296 ms
- 2 Queries
- 110 models

### CODE REVIEW

**Comment by nipunTharuksha** [comment]
**This should be the best answer with eloquent**

**Comment by nipunTharuksha** [comment]
sorry but this doesn't get **TOP 100 Movies** based on the rating. it gets 100 records and then order that records.

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -8,9 +8,16 @@ class HomeController extends Controller
 {
     public function index()
     {
-        $movies = Movie::all()->sortByDesc(function($movie) {
-            return $movie->ratings->avg('rating');
-        })->take(100);
+
+
+        $movies = Movie::query() //Using query for the next functions to be indented
```
Comment (SagarNaliyapara) [inline app/Http/Controllers/HomeController.php:13]: I really really like that, I've been using `query()` as well, I'm glad I'm not alone @Thotsuya :)

**Comment by PovilasKorop** [comment]
@Thotsuya if I'm not mistaken, your result is incorrect: you're taking only 100 movies, and only then sort them by rating. So you get 100 records sorted, not all movies sorted.

**Pull request closed without merging**